### PR TITLE
All source files in one branch

### PR DESCRIPTION
**Pull request by gabejohnson**
See #132

We would like to keep all source files in a single branch and use `gh-pages` purely for hosting
build artifacts.

Going forward we would build the PS source and push it (along with the html, css, other js dependencies, README.md, etc.) to `gh-pages`. No edits would happen directly in `gh-pages`.

So far I've done the minimum to move the frontend files into the mainline branch and get the Core backend working with `v0.13.6`. Much of the work was already done by @natefaubion 
 in #127 which I've merged into this branch. The other backends still need updating and there are none of the scripts mentioned in https://github.com/purescript/trypurescript/issues/132#issue-544413955.

I've forked the Core gist and the example gists to which it points to get them working with `Effect` and pointed `Try.API` to that fork. Going forward it may make sense to have those gists live under the `purescript` org.

I'm open to adding more to this PR (backend updates, build scripts) but wanted to keep it minimal and spur discussion concerning what needs to be done to get https://try.purescript.org up-to-date.

### CODE REVIEW

File: client/js/index.js [hunk 1]
```diff
@@ -0,0 +1,6231 @@
+// Generated by purs bundle 0.12.0
```
Comment (hdgarrood) [inline client/js/index.js]: Should this file be ignored?

**Comment by hdgarrood** [comment]
I'm very much in favour of this PR minimal, by the way. :+1:

File: client/js/index.js (hunk 1 above)
Comment (gabejohnson) [reply client/js/index.js]: I had just added a it to `.gitignore` right before opening the PR, but forgot to remove it. Fixed.

**Comment by citizengabe** [comment]
Those all sound like reasonable changes. I think replacing the `thermite` backend with `react-basic` also makes sense while updating the other backends.

Since I incorporated the changes from https://github.com/purescript/trypurescript/pull/127, that can probably be closed as well.

**Comment by gabejohnson** [comment]
Those all sound like reasonable changes. I think replacing the `thermite` backend with `react-basic` also makes sense while updating the other backends.

Since I incorporated the changes from https://github.com/purescript/trypurescript/pull/127, that can probably be closed as well.

**Comment by natefaubion** [comment]
I don't think it's worth updating backends. I would much rather merge up https://github.com/purescript/trypurescript/pull/128 and obviate the need for backends.

**Comment by gabejohnson** [comment]
> I would much rather merge up #128 and obviate the need for backends.

That makes sense. I guess just updating some gists then.

**Pull request merged**